src/lib/cli/format.ts

```diff
@@ -241,3 +241,37 @@ export const formatBytes = ( bytes: number, decimals = 2 ): string => {
 		sizes[ idx ]
 	}`;
 };
+
+/**
+ * Get the duration between two dates
+ *
+ * @param {Date} from The start date
+ * @param {Date} to  The end date
+ * @returns {string} The duration between the two dates
+ */
+export function formatDuration( from: Date, to: Date ): string {
+	const millisecondsPerSecond = 1000;
+	const millisecondsPerMinute = 60 * millisecondsPerSecond;
+	const millisecondsPerHour = 60 * millisecondsPerMinute;
+	const millisecondsPerDay = 24 * millisecondsPerHour;
+
+	const duration = Math.abs( from.getTime() - to.getTime() );
```

Comment:
We probably want to set the duration to 0 if it's negative due to race conditions or similar. And maybe ask the user to retry in 5 seconds?